src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -107,18 +115,27 @@ def _extract_messages(self) -> None:
                     exc_info=True,
                 )
                 continue
-
             try:
                 if self.__input_codec:
                     self.__input_codec.validate(parsed_payload)
             except ValidationError:
                 # For now while this is still experimental, those errors are
                 # not supposed to be fatal.
-                logger.warn(
+                logger.warning(
                     "process_messages.invalid_schema",
                     extra={"payload_value": str(msg.payload.value)},
                     exc_info=True,
                 )
+            if use_case_id := extract_use_case_id(parsed_payload["name"]):
+                parsed_payload["use_case_id"] = use_case_id
+            else:
+                logger.warning(
+                    "process_messages.invalid_metric_resource_identifier",
+                    extra={"payload_value": str(msg.payload.value)},
+                    exc_info=True,
+                )
+                continue
```

Comment:
I'm not sure we should continue if we don't recognize the namespace. This seems like a hard failure case to me (or at least a `logger.error`)

Reply:
For now it skips the current message it's processing and moves on to the next one. Updated the logger to error.